data/transactions/logic/eval.go

```diff
@@ -680,11 +681,9 @@ func eval(program []byte, cx *EvalContext) (pass bool, err error) {
 
 	defer func() {
 		// Ensure we update the debugger before exiting
-		if cx.Debugger != nil {
-			errDbg := cx.Debugger.Complete(cx.refreshDebugState(err))
-			if err == nil {
-				err = errDbg
-			}
+		errDbg := callAfterAppEvalHookIfItExists(cx.Debugger, cx.refreshDebugState(err))
```

Comment:
One downside of this style of method invocation: `cx.refreshDebugState` will be called even if `cx.Debugger` is nil, or if `cx.Debugger` does not implement this specific method. `cx.refreshDebugState` copies the entire stack and scratch space, so this is potentially an expensive operation to invoke frequently.